lib/galaxy/web/proxy/js/lib/proxy.js

```diff
@@ -119,9 +127,15 @@ DynamicProxy.prototype.handleWs = function(req, res, head) {
     this.rewriteRequest(req);
     res.oldWriteHead = res.writeHead;
     res.writeHead = function(statusCode, headers) {
-        res.setHeader('Access-Control-Allow-Origin', origin);
-        res.setHeader('Access-Control-Allow-Credentials', 'true');
-        res.oldWriteHead(statusCode, headers);
+        try {
+            res.setHeader('Access-Control-Allow-Origin', origin);
+            res.setHeader('Access-Control-Allow-Credentials', 'true');
+            res.oldWriteHead(statusCode, headers);
+        }
+        catch (error) {
+          console.log("Header could not be modified.");
+          console.log(error);
```

Comment:
```
Proxy error:  { [Error: connect ECONNREFUSED]
  code: 'ECONNREFUSED',
  errno: 'ECONNREFUSED',
  syscall: 'connect' }

/home/bag/projects/code/galaxy/lib/galaxy/web/proxy/js/lib/proxy.js:122
        res.setHeader('Access-Control-Allow-Origin', origin);
            ^
TypeError: Object #<Socket> has no method 'setHeader'
    at Socket.res.writeHead (/home/bag/projects/code/galaxy/lib/galaxy/web/proxy/js/lib/proxy.js:122:13)
    at /home/bag/projects/code/galaxy/lib/galaxy/web/proxy/js/lib/proxy.js:130:13
    at ClientRequest.onOutgoingError (/home/bag/projects/code/galaxy/lib/galaxy/web/proxy/js/node_modules/http-proxy/lib/http-proxy/passes/ws-incoming.js:123:9)
    at ClientRequest.EventEmitter.emit (events.js:95:17)
    at Socket.socketErrorListener (http.js:1547:9)
    at Socket.EventEmitter.emit (events.js:95:17)
    at net.js:441:14
    at process._tickCallback (node.js:415:13)
```
I think this will `fix` https://github.com/galaxyproject/galaxy/issues/1204
At least it will not crash the proxy.